libraries/chef_f5.rb

```diff
@@ -41,6 +44,33 @@ def pool_is_missing_node?(pool, node)
     members.map { |m| m[:address] }.grep(/#{with_partition node}/).empty?
   end
 
+  def pool_is_missing_monitor?(pool, monitor)
+    monitors = api.LocalLB.Pool.get_monitor_association(pool_names:
+      {item: with_partition(pool)}
+    )[:item]
+
+    monitors = [ monitors ] if monitors.is_a? Hash
+    monitors.select do |mon|
+      mon[:monitor_rule][:monitor_templates][:item] == with_partition(monitor)
+    end.empty?
+  end
+
+  def add_monitor(pool, monitor)
+    api.LocalLB.Pool.set_monitor_association(monitor_associations: {
+      item: [
+        {pool_name: pool,
+         monitor_rule: {
+           monitor_templates: {
+             item: monitor
+           }
+         },
+         quorum: '0',
+         type: 'MONITOR_RULE_TYPE_SINGLE'
```

Comment:
Hm I hadn't considered multiple monitors. Does this break if the pool already has another monitor?

I don't think this PR needs to address the more complicated scenarios but I just want to make sure it's not going to break existing stuff.